cdr_product_change_runs.py

```
@@ -110,30 +110,34 @@ def _sqlite_facts_compatible(path: Path, expected_run_date: str) -> bool:
     database = export_root / "local-cdr.sqlite"
     if not _sqlite_facts_compatible(database, run_date(export_root)):
-        return None
+        return
     selected = ", ".join(f'"{column}"' for column in _FACT_COLUMNS)
-    products: Dict[str, Dict[str, Any]] = {}
     with sqlite3.connect(_sqlite_uri(database), uri=True) as connection:
         connection.row_factory = sqlite3.Row
         rows = connection.execute(
             f"SELECT {selected} FROM bank_product_facts "
-            "ORDER BY provider COLLATE NOCASE, product_id, dataset, fact_id"
+            "ORDER BY provider COLLATE NOCASE, provider, product_id, dataset, fact_id"
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)